test/dict.jl

```diff
@@ -787,6 +787,15 @@ end
           [v for (k, v) in d] == [d[x[1]] for (i, x) in enumerate(d)]
 end
 
+@testset "consistency of dict iteration order (issue #56841)" begin
+    dict = Dict(randn() => randn() for _ = 1:100)
+    foreach(zip(dict, keys(dict), values(dict), pairs(dict))) do (d, k, v, p)
+        @test d == p
+        @test first(d) == first(p) == k
+        @test last(d) == last(p) == v
+    end
```

Comment:
In case there's a bug that decouples `keys`/`values`/`iterate`/`pairs`, this will cause 300 individual failures, each showing up in the log. I think it'd be better for debugging to only have one failure through `@test all(...)`, instead of three `@test` per key-value-pair in the dict.